### DISCUSSION TITLE
Remote user's avatar

### DISCUSSION THREAD

**Discussion by ogurec-ogurec** [danirus/django-comments-xtd#268] [Q&A]
I have successfully installed the avatar by [the guide readthedocs](https://django-comments-xtd.readthedocs.io/en/latest/usecases/change-user-image-or-avatar.html?highlight=avatar#add-django-avatar-to-the-comp-project).

The situation is as follows: I removed everywhere (django-comments-xtd and django-avatar) the gravatar function. I display the avatars that the user has uploaded. Everything works well, but there is such a nuance, if I delete a user, an empty avatar is displayed at the comment of the deleted user. How can I check that the user is deleted, and assign, for example, default-deleted.jpg?

Avatars display in templates this:

`   <img src = "{% avatar_url comment.user%}"  />`

![image](https://user-images.githubusercontent.com/66796268/105646022-c74bb200-5e30-11eb-98c6-5670e0f0a4cc.png)

Bob Jenkins - is a remote user.

**Comment by danirus** [comment]
When you delete the user, the foreign key `user` field in the comments table becomes `null`.
That's the effect of the `SET_NULL` [here](https://github.com/django/django-contrib-comments/blob/master/django_comments/abstracts.py#L53) in django-contrib-comments.

I can figure that the blank avatar is the effect of using the `avatar_url` template tag with the `comment.user` being `None`. I don't know how to modify that behavior. I would recommend you to look in its settings, maybe using [this one](https://django-avatar.readthedocs.io/en/latest/#AVATAR_GRAVATAR_DEFAULT) could give you the solution. 
Let us know if you find the solution.

**Reply by ogurec-ogurec** [reply]
@ danirus thanks for the hint, I found a solution.

In the file **providers.py**  django-avatar lib: 
path: `/Lib/site-packages/avatar/providers.py:`

```
class PrimaryAvatarProvider (object):

     @classmethod
     def get_avatar_url (self, user, size):
       #Add path media deleted-avatar 
       path_deleted_avatar = getattr (settings, 'MEDIA_AVATAR_DELETED', '')
       #Add deleted image
       deleted_image = 'deleted.png'

       avatar = get_primary_avatar (user, size)
       if avatar:
           return avatar.avatar_url (size)
       #if we get None instead of avatar
       elif avatar is None:
           return '%s%s' (path_deleted_avatar, deleted_image)
```

Result :) 

![image](https://user-images.githubusercontent.com/66796268/106148589-8b546d80-610c-11eb-91dc-a52ad58b635b.png)

**Reply by ogurec-ogurec** [reply]
Now there is a new idea, but what if you assign a first name and last name to a remote user, for example - deleted deleted?
